### PR TITLE
Send push from backend

### PR DESCRIPTION
**Pull request by Kalk88**
## **Description**

Send push notifications to users who's schedules are in range of the time of execution of the
function. Then update the schedules next time to water with the stored interval.

## **How to test?**

Create a schedule. 
Register a device token manually in the database.
curl the the endpoint.

### CODE REVIEW

File: server/cloud-functions/functions/index.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      console.log(`Found ${snapshot.docs.length} schedules to update`)
+      return snapshot.docs
+        .map(doc => ({
+          id: doc.id,
+          schedule: doc.data()
+        }))
+        .reduce(reduceSchedulesOnUserId, {})
+    })
+    .then(reduced => {
+      return Promise.all(Object.entries(reduced).map(([userId, data]) => {
```
Comment (PiddePannkauga) [inline server/cloud-functions/functions/index.js]: Maybe use [this](https://stackoverflow.com/questions/30362733/handling-errors-in-promise-all) instead

Comment (PiddePannkauga) [reply server/cloud-functions/functions/index.js]: Did you take action on this?

Comment (Kalk88) [reply server/cloud-functions/functions/index.js]: @PiddePannkauga Not yet

**Comment by Kalk88** [comment]
We are going to do this in typescript and in two steps, update schedules and send notifications

**Pull request closed without merging**